nixos/modules/services/hardware/pid-fan-controller.nix

```
@@ -0,0 +1,190 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.pid-fan-controller;
+  heatSource = {
+    options = {
+      name = lib.mkOption {
+        type = lib.types.uniq lib.types.nonEmptyStr;
+        description = "Name of the heat source";
+      };
+      wildcardPath = lib.mkOption {
+        type = lib.types.nonEmptyStr;
+        description = ''
+          Path of the heat sources hwmon 'temp_input' file.
+          This path can contain multiple wildcards, but has to resolve to
+          exactly one result.
+        '';
+      };
+      pidParams = {
+        setPoint = lib.mkOption {
+          type = lib.types.ints.unsigned;
+          description = "Set point of the controller in °C";
+        };
+        P = lib.mkOption {
+          description = "K_p of PID controller";
+          type = lib.types.float;
+        };
+        I = lib.mkOption {
+          description = "K_i of PID controller";
+          type = lib.types.float;
+        };
+        D = lib.mkOption {
+          description = "K_d of PID controller";
+          type = lib.types.float;
```

```suggestion
        P = lib.mkOption {
          description = "K_p of the PID controller.";
          type = lib.types.float;
        };
        I = lib.mkOption {
          description = "K_i of the PID controller.";
          type = lib.types.float;
        };
        D = lib.mkOption {
          description = "K_d of the PID controller.";
          type = lib.types.float;
```
Ditto for this. If `K_*` is obvious to users, leave as it is; otherwise, explain it.